server/src/main.rs

```diff
@@ -1,3 +1,177 @@
+extern crate hyper;
+extern crate reroute;
+extern crate serde;
+extern crate serde_json;
+
+//structs found in /serde_types.in.rs
+//need this line to make serialization of structs work.
+include!(concat!(env!("OUT_DIR"), "/serde_types.rs"));
+
+use std::*;
+use hyper::server::{Server, Request, Response};
+use hyper::status::StatusCode;
+use reroute::{Captures, Router};
+
+//static VERSION : &'static str = "/api/v1/";
+
+//REST ENDPOINTS
+static FILE_SYS_END_PT : &'static str = r"/api/v1/filesystem/[[:ascii:]]*";
+static PROCESS_END_PT : &'static str = "/api/v1/processes/";
+static DEBUG_ATCH_PID_END_PT : &'static str = r"/api/v1/debug/attach/pid/[[:digit:]]*";
+static DEBUG_ATCH_BIN_END_PT : &'static str = r"/api/v1/debug/attach/bin/[[:ascii:]]*";
+static DEBUG_END_PT : &'static str = "/api/v1/debug";
+static DEBUG_FUNC_LST_END_PT : &'static str = "/api/v1/debug/functions";
+static DEBUG_FUNC_END_PT : &'static str = r"/api/v1/debug/functions/[[:ascii:]]*";
+static BRKPNT_END_PT : &'static str = "/api/v1/debug/breakpoints";
+static FUNC_BRKPNT_END_PT : &'static str = r"/api/v1/debug/breakpoints/[[:ascii:]]*"; //PUT AND DELETE
+static DEBUG_EXEC_PROC_END_PT : &'static str = "/api/v1/debug/execute";
+static DEBUG_EXEC_FUNC_END_PT : &'static str = r"/api/v1/debug/functions/[[:ascii:]]*/execute";
+
+static DEBUG_EXEC_LST_END_PT : &'static str = "/api/v1/debug/executions";
+static DEBUG_EXEC_STATUS_END_PT : &'static str = r"/api/v1/debug/executions/[[:digit:]]*";
+static DEBUG_EXEC_TRACE_END_PT : &'static str = r"/api/v1/debug/executions/[[:digit:]]*/trace";
+static DEBUG_EXEC_STOP_END_PT : &'static str = r"/api/v1/debug/executions/[[:digit:]]*/stop";
+
+// @ /greet
+fn basic_handler(_: Request, res: Response, c: Captures){
+    println!("captures: {:?}", c);
+    res.send(b"He who controls the spice...").unwrap();
+}
+
+// @ /filesystem/:path*
+fn filesystem_handler(_: Request, res: Response, c: Captures){
+    println!("file system captures captures: {:?}", c); //debug line
+
+    let c_str = &c.unwrap()[0];
+    
+    println!{"c_str value: {}", c_str}; //debug line
+    
+    let path = c_str.split("/api/v1/filesystem").nth(1).unwrap();
+
+    println!("path value: {}", path);// debug line
+    
+    let file_meta = fs::metadata(path).unwrap();
+
+    let mut file_type = "".to_string();
+
+    if file_meta.is_dir(){
+        file_type = "dir".to_string();
+    } else{
+        file_type = "file".to_string();
+    }
+
+    let my_file = File{name: path.to_string(), path: path.to_string(), f_type: file_type, contents: "Hello World".to_string()};
```

Comment:
Could split this line up to keep it under 100 characters and to give some more spacing